lib/iris/analysis/maths.py

```diff
@@ -120,29 +120,25 @@ def _assert_is_cube(cube):
 def _assert_compatible(cube, other):
```

Comment:
Although, seeing as da.broadcast_to is only used to validate that the cubes are compatible da.core.broadcast_shapes is potentially a more efficient method as it seems to do less work.